From 17b6881fc25f6bfc5a8e78de4f070dc7af4fff42 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 6 Jun 2012 16:33:45 +0800
Subject: [PATCH 285/486] ENGR00212563 Change camera switch ioctl to Camera
 device node with camera ID

Align the kernel's change in dual camera support
/dev/video0 for camera 0
/dev/video1 for camera 1

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 mx6/libcamera/Android.mk        |  2 +-
 mx6/libcamera/V4l2CsiDevice.cpp | 23 ++++++++++++++++++++++-
 2 files changed, 23 insertions(+), 2 deletions(-)

diff --git a/mx6/libcamera/Android.mk b/mx6/libcamera/Android.mk
index cea4761..a755290 100755
--- a/mx6/libcamera/Android.mk
+++ b/mx6/libcamera/Android.mk
@@ -65,7 +65,7 @@ else
 endif
 
 #Define this for switch the Camera through V4L2 MXC IOCTL
-LOCAL_CPPFLAGS += -DV4L2_CAMERA_SWITCH
+#LOCAL_CPPFLAGS += -DV4L2_CAMERA_SWITCH
 
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw	
 LOCAL_MODULE:= camera.$(TARGET_BOARD_PLATFORM)
diff --git a/mx6/libcamera/V4l2CsiDevice.cpp b/mx6/libcamera/V4l2CsiDevice.cpp
index 9a61caf..0f8da8a 100755
--- a/mx6/libcamera/V4l2CsiDevice.cpp
+++ b/mx6/libcamera/V4l2CsiDevice.cpp
@@ -64,7 +64,9 @@ namespace android{
 
         if(mCameraDevice > 0)
             return CAPTURE_DEVICE_ERR_ALRADY_OPENED;
-        else if (mCaptureDeviceName[0] != '#'){
+
+#ifdef V4L2_CAMERA_SWITCH
+        if (mCaptureDeviceName[0] != '#'){
             CAMERA_HAL_LOG_RUNTIME("already get the device name %s", mCaptureDeviceName);
             mCameraDevice = open(mCaptureDeviceName, O_RDWR, O_NONBLOCK);
             if (mCameraDevice < 0)
@@ -120,6 +122,25 @@ namespace android{
                 return CAPTURE_DEVICE_ERR_OPEN;
             }
         }
+#else
+        memset((void *)dev_node, 0, CAMAERA_FILENAME_LENGTH);
+        sprintf(dev_node, "/dev/video%d", cameraId);
+        if ((fd = open(dev_node, O_RDWR, O_NONBLOCK)) < 0) {
+            CAMERA_HAL_ERR("dev_node %s:cannot be opened", dev_node);
+            return CAPTURE_DEVICE_ERR_OPEN;
+        }
+
+        if(ioctl(fd, VIDIOC_DBG_G_CHIP_IDENT, &vid_chip) < 0 ) {
+            close(fd);
+            CAMERA_HAL_ERR("dev_node %s:cannot get sensor name", dev_node);
+            return CAPTURE_DEVICE_ERR_OPEN;
+        }
+        strcpy(mCaptureDeviceName, dev_node);
+        strcpy(mInitalDeviceName, vid_chip.match.name);
+        CAMERA_HAL_LOG_INFO("device name is %s", mCaptureDeviceName);
+        CAMERA_HAL_LOG_INFO("sensor name is %s", mInitalDeviceName);
+        mCameraDevice = fd;
+#endif
         return ret; 
     }
 
-- 
1.8.0

